cmd/ignite/run/create_test.go

```diff
@@ -0,0 +1,204 @@
+				meta.PortMapping{
+					BindAddress: net.IPv4(0, 0, 0, 0),
+					HostPort:    uint64(80),
+					VMPort:      uint64(80),
+					Protocol:    meta.ProtocolTCP,
+				},
+			},
+		},
+		{
+			name: "invalid port mapping",
+			createFlag: &CreateFlags{
+				VM:           &api.VM{},
+				PortMappings: []string{"1.1.1.1:foo:bar"},
+			},
+			args: []string{testImage},
+			err:  true,
+		},
+		{
+			name: "ssh public key set",
+			createFlag: &CreateFlags{
+				VM: &api.VM{},
+				SSH: api.SSH{
+					Generate:  true,
+					PublicKey: "some-pub-key",
+				},
+			},
+			args: []string{testImage},
+			wantSSH: api.SSH{
+				Generate:  true,
+				PublicKey: "some-pub-key",
+			},
+		},
+		{
+			name: "with no VM name and --require-name flag set",
+			createFlag: &CreateFlags{
+				VM:          &api.VM{},
+				RequireName: true,
+			},
+			args: []string{testImage},
+			err:  true,
+		},
+	}
+
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			err := rt.createFlag.constructVMFromCLI(rt.args)
+			if (err != nil) != rt.err {
+				t.Errorf("expected error %t, actual: %v", rt.err, err)
+			}
+
+			if !rt.err {
+				// Check if the VM image is set as expected.
+				if rt.createFlag.VM.Spec.Image.OCI != testOCIRef {
+					t.Errorf("expected VM.Spec.Image.OCI to be %q, actual: %q", testOCIRef.String(), rt.createFlag.VM.Spec.Image.OCI.String())
+				}
+
+				// Check if copy files are set as expected.
+				if len(rt.wantCopyFiles) > 0 {
+					if !reflect.DeepEqual(rt.createFlag.VM.Spec.CopyFiles, rt.wantCopyFiles) {
+						t.Errorf("expected VM.Spec.CopyFiles to be %v, actual: %v", rt.wantCopyFiles, rt.createFlag.VM.Spec.CopyFiles)
+					}
+				} else {
+					// If the copy files map is empty, compare their sizes.
+					if len(rt.wantCopyFiles) != len(rt.createFlag.VM.Spec.CopyFiles) {
+						t.Errorf("expected VM.Spec.CopyFiles to be %v, actual: %v", rt.wantCopyFiles, rt.createFlag.VM.Spec.CopyFiles)
+					}
+				}
+
+				// Check if port mappings are set as expected.
+				if reflect.DeepEqual(rt.createFlag.VM.Spec.Network.Ports, rt.wantPortMapping) {
+					t.Errorf("expected VM.Spec.Network.Ports to be %v, actual: %v", rt.wantPortMapping, rt.createFlag.VM.Spec.Network.Ports)
+				}
+
+				// Check if the SSH values are set as expected.
+				if reflect.DeepEqual(rt.createFlag.VM.Spec.SSH, rt.wantSSH) {
+					t.Errorf("expected VM.Spec.SSH to be %v, actual: %v", rt.wantSSH, rt.createFlag.VM.Spec.SSH)
+				}
+			}
+		})
+	}
+}
+
+func TestNewCreateOptions(t *testing.T) {
+	tests := []struct {
+		name       string
+		createFlag *CreateFlags
+		err        bool
+	}{
+		{
+			name: "require-name with no name",
+			createFlag: &CreateFlags{
+				VM:          &api.VM{},
+				RequireName: true,
+			},
+			err: true,
+		},
+		{
+			name: "require-name with VM config",
+			createFlag: &CreateFlags{
+				ConfigFile:  fmt.Sprintf("testdata%c%s", filepath.Separator, "input/create-config-no-name.yaml"),
+				RequireName: true,
+			},
+			err: true,
+		},
```

Comment:
These cases cover only the failure cases of `NewCreateOptions`. Success case would lead to actual image imports and would require setting up a back-end storage as well. These tests are enough for checking the `--require-name` flag.